### PR TITLE
RSDK-8436: Take a swing at fixing peerconnection close problems.

### PR DESCRIPTION
**Pull request by dgottlieb**

### CODE REVIEW

**Comment by dgottlieb** [comment]
The test failures are unrelated to this patch. I filed [this ticket](https://viam.atlassian.net/browse/RSDK-8480).

File: rpc/wrtc_base_channel.go [hunk 1]
```diff
@@ -109,7 +109,7 @@ func newBaseChannel(
 				//
 				// We chose here to call close for all cases of `Disconnected`, `Failed` and
 				// `Closed`. We rely on pion's `PeerConnection.Close` method being idempotent.
```
Comment (benjirewis) [inline rpc/wrtc_base_channel.go:111]:
```suggestion
				// `Closed`. We rely on pion's `PeerConnection.GracefulClose` method being idempotent.
				// We "gracefully" close to wait for the read loop to complete.
```
At least one mention of the `Graceful` usage might be helpful.

**Review by benjirewis — COMMENTED** [review]
There's no place we want to `Close` a peer connection without waiting for read activity to finish?

**Comment by dgottlieb** [comment]
> There's no place we want to Close a peer connection without waiting for read activity to finish?

Certainly Close not waiting for goroutines to exit makes us susceptible to goroutine leaks. I'm happy to entertain suggestions!

**Review by benjirewis — APPROVED** [review]
No, I'm not sure what I was thinking haha; `GracefulClose` seems best in all of these occurrences. LGTM.

**Pull request merged**